src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -232,19 +305,25 @@ public Authentication authenticate(Authentication authentication) throws Authent
     }
 
     protected OAuthSession newOAuthSession(String from, final String redirectOnFinish) {
-    	final GenericUrl tokenServerURL = new GenericUrl(serverPrefix + "/oauth/token");
-        final String authorizationServerURL = serverPrefix +"/oauth/authorize";
+    	LOGGER.fine(String.format("newOAuthSession from %s redirectOnFinish %s", from, redirectOnFinish));
+    	final GenericUrl tokenServerURL = new GenericUrl(getServerPrefix(overrides) + "/oauth/token");
+        final String authorizationServerURL = getServerPrefix(overrides) +"/oauth/authorize";
+        
+        LOGGER.fine(String.format("tokenServerURL %s auth server url %s clientid %s clientSecret %s transport %s", tokenServerURL, authorizationServerURL, getClientId(overrides), getClientSecret(overrides) == null ? null : getClientSecret(overrides).getPlainText(), transport == null ? null : transport.toString()));
         
         final AuthorizationCodeFlow flow = new AuthorizationCodeFlow.Builder(
                 BearerToken.queryParameterAccessMethod(), transport, JSON_FACTORY, tokenServerURL,
-                new ClientParametersAuthentication(clientId, clientSecret.getPlainText()), clientId, authorizationServerURL)
+                new ClientParametersAuthentication(getClientId(overrides), getClientSecret(overrides).getPlainText()), getClientId(overrides), authorizationServerURL)
```

Comment:
IMO, there is not a truly common pattern in Jenkins for this.  There is some "macro" related plugins wrt jenkinst global/job env vars and build steps (where there could be multiple instances of these build steps), but I have not seen much in the way of a common pattern where you discern defaults outside of those specific scopes.

The map based pattern for parameter storage stems from conventions on previous teams I was on ... old habits :-) ...

I'm good with moving to specific gets ... will move to that.